tapiro/core/src/main/scala/io/buildo/tapiro/AkkaHttpMeta.scala

```diff
@@ -40,17 +41,26 @@ object AkkaHttpMeta {
     q"pathPrefix($pathName) { List(..$rest).foldLeft[Route]($first)(_ ~ _) }"
   }
 
-  val endpoints = (routes: List[Route]) =>
-    routes.flatMap { route =>
-      val name = Term.Name(route.name.last)
-      val endpointsName = Term.Select(Term.Name("endpoints"), name)
-      val controllersName = Term.Select(Term.Name("controller"), name)
+  val endpoints = (routes: List[TapiroRoute]) =>
+    routes.map { route =>
+      val name = Term.Name(route.route.name.last)
+      val endpointsName = q"endpoints.$name"
+      val controllersName = q"controller.$name"
       val controllerContent =
-        if (route.params.length <= 1) Some(controllersName)
-        else Some(Term.Select(Term.Eta(controllersName), Term.Name("tupled")))
-      controllerContent.map { content =>
-        val toRoute = Term.Apply(Term.Select(endpointsName, Term.Name("toRoute")), List(content))
-        q"val ${Pat.Var(name)} = $toRoute"
-      }
+        route.method match {
+          case RouteMethod.GET =>
+            route.route.params.length match {
+              case 0 => q"_ => $controllersName()"
+              case 1 => controllersName
+              case _ => q"($controllersName _).tupled"
+            }
+          case RouteMethod.POST =>
+            val fields = route.route.params
+              .filterNot(_.tpe == MetarpheusType.Name("AuthToken"))
```

Comment:
Is it correct that we hardcode `AuthToken` here?

Reply:
I think it's sort-of-correct right now because we're hardcoding it everywhere anyway, so we expect the type parameter in the controller trait to be always named exactly `AuthToken`. As I understand it, we should fix that in #204 when we change metarpheus to return the real name of the type parameter